pkg/cache/lrucache.go

```diff
@@ -105,7 +105,7 @@ func (c *GitRepoLRUCache) Get(key string) *GitRepoFilePath {
 // Unlocking the cache is done manually (and not through "defer c.lock.Unlock()"
 // in order to free other threads to perform cache operations when possibly
 // lengthy git cloning operations are being performed on individual elements.
-func (c *GitRepoLRUCache) Put(key string) (*GitRepoFilePath, error) {
+func (c *GitRepoLRUCache) Put(key string, neverEvictRepos map[string]bool) (*GitRepoFilePath, error) {
```

Comment:
I think the API for `Put` shouldn't change.

Having to provide a list of repos that won't be evicted from the cache each time pollutes it abit and I think instead, we can add this as a data member to the cache itself:

```go
struct GitRepoLRUCache struct {
    ...
    
    neverEvict map[string]bool
}
```

That way, it's a simple matter of accessing the data member of the LRU cache and checking it. This also helps prevent a case where a consumer of the LRU cache attempts to drop in different lists of `neverEvictRepos` each time.